lib/galaxy/config/__init__.py

```diff
@@ -274,13 +274,17 @@ def convert_datatype(key, value):
             datatype = self.schema.app_schema[key].get('type')
             # check for `not None` explicitly (value can be falsy)
             if value is not None and datatype in type_converters:
-                return type_converters[datatype](value)
+                # convert value or each item in value to type `datatype`
+                if type(value) is list:
+                    return [type_converters[datatype](item) for item in value]
```

Comment:
Assigning `type_converters[datatype]` to a variable (before the `if`) would make this more readable, I think.

Reply:
Agreed. Done.